xmppserver/src/main/java/org/jivesoftware/admin/servlet/SystemCacheDetailsServlet.java

```diff
@@ -46,20 +46,23 @@ protected void doGet(final HttpServletRequest request, final HttpServletResponse
             request.setAttribute("warningMessage", LocaleUtils.getLocalizedString("system.cache-details.cache_not_found", Collections.singletonList(cacheName)));
         }
 
+        final boolean secretKey = optionalCache.map(Cache::isKeySecret).orElse(Boolean.FALSE);
+        final boolean secretValue = optionalCache.map(Cache::isValueSecret).orElse(Boolean.FALSE);
+
         final List<Map.Entry<String, String>> cacheEntries = optionalCache.map(Cache::entrySet)
             .map(Collection::stream)
             .orElseGet(Stream::empty)
-            .map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey().toString(), entry.getValue().toString()))
+            .map(entry -> new AbstractMap.SimpleEntry<>(secretKey ? "************" : entry.getKey().toString(), secretValue ? "************" : entry.getValue().toString()))
```

Comment:
I have no experience with swapping out SimpleEntry instances like this. How does this behave if a map has more than one secret key? I'm worried that both keys are replaced by `************`, which then is a duplicate key, causing map entries to disappear.

Reply:
What's being worked on is a list It's a list of map entries, so duplicate key's should not be a problem.